simcon_project/users/views/user_registration.py

```diff
@@ -0,0 +1,54 @@
+from django.shortcuts import render, redirect
+from users.models import Student
+from django.contrib.auth import login
+from users.forms import NewStudentCreationForm
+from django.contrib import messages
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
+
+
+def UserRegistration(request, uidb64):
+        if request.method == "POST":
+                form = NewStudentCreationForm(request.POST)
+
+                if form.is_valid():
+                        password1 = form.cleaned_data.get('password1')
+                        password2 = form.cleaned_data.get('password2')
+
+                        if password1 == password2:
+                                email = form.cleaned_data.get('email')
+
+                                #if the user is created and not registered
+                                if Student.objects.filter(email=email, registered=False):
+                                        #find the students account
+                                        user = Student.objects.get(email=email, registered=False)
+
+                                        #if the uid from the email matches the students uid, then edit user with input
+                                        if uidb64 == urlsafe_base64_encode(force_bytes(user.pk)):
+                                                user = Student.objects.get(email=email, registered=False)
+                                                user.set_password(form.cleaned_data.get('password1'))
+                                                user.first_name = form.cleaned_data.get('first_name')
+                                                user.last_name = form.cleaned_data.get('last_name')
+                                                user.registered = True
+                                                user.save()
+                                                login(request, user)
+                                                return redirect('/student-view/')  #sends to the student view after completion
```

Comment:
Curious what people think about redirecting straight to the student view? I redirected to the login page for the researcher registration but I'm not sure if that provides any benefit over going straight to their normal view.